### DISCUSSION TITLE
Looking for ways to decrease run times.

### DISCUSSION THREAD

**Discussion by Kazagan** [QuestPDF/QuestPDF#842] [General]
I'm generating a document for a Client, where I'm creating many  tables for different data points. 

Something like:

##### data point
| Header A | Header B | Header C | Header D |
| ----------| --------- | ----------| ---------- | 
| row 1        | data          | data         | data          |
| row 2        | data          | data         | data          |
| row 3        | data          | data         | data          |
| row 4        | data          | data         | data          |

Tables like this with different number of rows, and different number of columns are generated in a manual configuration, where sometimes there will be a single column of tables allowing it to take up the full screen, or 2 columns of tables. Requiring manual measurements of the size of the table to handle paging, and splitting up of the tables. The table instead of paging onto the next page could page into the next column.
To do the measurements, the above represents defined class that gets measured based on the number of rows, and how much room the text fields take up, with manual measurements to keep it to 3 line or less of wrapping. 

For smaller data sets it runs great, but have had larger data sets with 10,000 of those tables, with a total number of rows around 49364 rows. 2053 pages

It takes around 4 minutes to generate the document. I attempted to time out how long the manual measurements were taking to see if the problem was with that, but the measurements weren't taking up a significant amount of time.

**Comment by MarcinZiabek** [comment]
Please share the method of how the table is created. What version of the library are you using?

**Comment by Kazagan** [comment]
Tables are created with the table element, using 2024.3.0.
Hope that's the information you were looking for, but can answer further if needed

**Reply by MarcinZiabek** [reply]
Would you please share the code that creates the table? A screenshot would also be helpful.

**Comment by Kazagan** [comment]


**Reply by Kazagan** [reply]
Data on the pdf looks like nonsense, because this is from some unit tests that use randomly generated data to validate that the pdf can generate under different configuration settings. The image is our most common configuration, with some stuff stripped out, such as page header and footers. 
The tables are rendered in a news paper format, going 
1 3
2 4

**Reply by MarcinZiabek** [reply]
I am sorry for the delayed response. Have you profiled your code? Would you please share what parts of it are taking the most time? I am trying to detect where exactly the problem can be and where we can improve performance.

**Reply by Kazagan** [reply]
I Have, it appears to slow down once I call the .Compose, After all data has been loaded, and all quest pdf configuration has been completed as well. I'm preparing a more in depth look at the full code to create the pdf as well in case that will help. Working with client to ensure it's good to share first. 

Sorry for my delayed response here.

**Comment by Kazagan** [comment]
Is there a good email address I can send you a zip file?

**Reply by MarcinZiabek** [reply]
Please feel free to use contact@questpdf.com 😄

**Reply by Kazagan** [reply]
Nevermind, The zip file was too large to send over email. I'm going to create a private git repo, is it okay If I add you as a contributor to look at it?

**Reply by MarcinZiabek** [reply]
@Kazagan Thank you for providing the sample repository. I have investigated the performance issue and sent my findings via email.

**Reply by Kazagan** [reply]
Apologies for leaving this open for so long, I went on vacation shortly after my last post, and forgot to return to this.
